src/fireo/queries/query_wrapper.py

```diff
@@ -19,6 +20,9 @@ def from_query_result(cls, model, doc, nested_doc=False):
         else:
             return None
 
+        # save the firestore reference doc so that further actions can be performed (i.e. collections())
+        model._meta.set_reference_doc(doc.reference)
+
```

Comment:
`doc` is not always `reference` object when it is a **nested model** then `doc` does not contain `reference` Actually **nested model** is python `dict`. **Nested Model** is not a regular **Model** 

Add **doc reference** when **doc** is not **nested model** you can place this code under [Here](https://github.com/octabytes/FireO/blob/bfd4e54e202a873c0e11a81361bf0d6603061f19/src/fireo/queries/query_wrapper.py#L55-L56)

Reply:
@AxeemHaider 👍